### PR TITLE
X-Site integration with Minikube use case

### PR DESCRIPTION
**Pull request by galderz**
X-Site implementation that works on Minikube.

This PR sets the foundation for the support of X-Site on OpenShift,
and provides us with the guarantees that the crucial elements work as expected.

A further PR will come in next sprint with the changes required to support OpenShift.
These are mostly related to service type and ports (LoadBalancer instead of NodePort),
and changes to authentication with remote OpenShift clusters (via username/password or token).

### CODE REVIEW

**Comment by galderz** [comment]
I'll do the necessary rebasing later today, just putting it for review for now.

@oraNod Note the changes in the design, the minikube example, and the README in the manual xsite minikube folder.

**Comment by galderz** [comment]
Rebased

File: pkg/controller/infinispan/infinispan_controller.go [hunk 1]
```diff
@@ -123,25 +127,34 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
...
 
+	// Check x-site configuration first.
+	// Must be done before creating any Infinispan resources,
+	// because remote site host:port combinations need to be injected into Infinispan.
+	reqLogger.Info("check if cross-site is configured")
+	xsite, err := r.computeXSite(infinispan, reqLogger)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	//xsite := &ispnutil.XSite{}
```
Comment (rigazilla) [inline pkg/controller/infinispan/infinispan_controller.go]: @galderz  I can merge this PR. Do you want to clean up this comment and squash the commits?

Comment (galderz) [reply pkg/controller/infinispan/infinispan_controller.go]: Yeah, let me cleanup and squash things first.

**Comment by rigazilla** [comment]
Thank you @galderz !

**Pull request merged**